tests/rptest/scale_tests/many_partitions_test.py

```diff
@@ -199,39 +201,36 @@ def __init__(self,
             self.expect_bandwidth /= 2
             self.expect_single_bandwidth /= 2
 
-        mb_per_partition = 1
+        # Clamp the node memory to exercise the partition limit.
+        mb_per_partition = 4
```

Comment:
You could make this a `@parameterize(mb_per_partition=4)` then it's easy to just run this test with different values.

But maybe this is something for later.